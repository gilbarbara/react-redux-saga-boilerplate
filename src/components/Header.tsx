import React from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { Container, responsive } from 'styled-minimal';

import { appColor, headerHeight, spacer } from 'modules/theme';

import { logOut } from 'actions';

import Icon from 'components/Icon';
import Logo from 'components/Logo';

const HeaderWrapper = styled.header`
  background-color: #113740;
  height: ${headerHeight}px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;

  &:before {
    background-color: ${appColor};
    bottom: 0;
    content: '';
    height: 0.2rem;
    left: 0;
    position: absolute;
    right: 0;
  }
`;

const HeaderContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  justify-content: space-between;
  padding-bottom: ${spacer(2)};
  padding-top: ${spacer(2)};
`;

const Logout = styled.button`
  align-items: center;
  color: #fff;
  display: flex;
  font-size: 1.3rem;
  padding: ${spacer(2)};

  ${responsive({ lg: { fontSize: '1.6rem' } })}; /* stylelint-disable-line */

  &.active {
    color: #fff;
  }

  span {
    display: inline-block;
    margin-right: 0.4rem;
    text-transform: uppercase;
  }
`;

export default function Header() {
  const dispatch = useDispatch();

  const handleClickLogout = () => {
    dispatch(logOut());
  };

  return (
    <HeaderWrapper data-testid="Header">
      <HeaderContainer>
        <Logo />
        <Logout onClick={handleClickLogout}>
          <span>logout</span>
          <Icon name="sign-out" width={16} />
        </Logout>
      </HeaderContainer>
    </HeaderWrapper>
  );
}
