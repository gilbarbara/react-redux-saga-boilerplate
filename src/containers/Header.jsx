import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import rgba from 'polished/lib/color/rgba';
import { appColor, headerHeight } from 'modules/theme';

import { logOut } from 'actions/index';

import { Container, utils } from 'styled-minimal';
import Icon from 'components/Icon';
import Logo from 'components/Logo';

const { responsive, spacer } = utils;

const HeaderWrapper = styled.header`
  background-color: ${rgba(appColor, 0.9)};
  height: ${headerHeight}px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;

  &:before {
    background-color: ${appColor};
    bottom: 0;
    content: '';
    height: 0.2rem;
    left: 0;
    position: absolute;
    right: 0;
  }
`;

const HeaderLogo = styled(Logo)`
  svg {
    width: 5rem;
  }
`;

const StyledContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  justify-content: space-between;
  padding-bottom: ${spacer(2)};
  padding-top: ${spacer(2)};
`;

const Logout = styled.button`
  align-items: center;
  color: #333;
  display: flex;
  font-size: 1.3rem;
  padding: ${spacer(2)};

  ${responsive({ lg: 'font-size: 1.6rem;' })}; /* stylelint-disable-line */

  &.active {
    color: #000;
  }

  span {
    display: inline-block;
    margin-right: 0.4rem;
    text-transform: uppercase;
  }
`;

export default class Header extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };

  handleClickLogout = () => {
    const { dispatch } = this.props;

    dispatch(logOut());
  };

  render() {
    return (
      <HeaderWrapper>
        <StyledContainer>
          <HeaderLogo />
          <Logout onClick={this.handleClickLogout}>
            <span>logout</span>
            <Icon name="sign-out" color="#333" width={16} />
          </Logout>
        </StyledContainer>
      </HeaderWrapper>
    );
  }
}
