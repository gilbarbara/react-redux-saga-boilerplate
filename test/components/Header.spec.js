import React from 'react';
import { mount } from 'enzyme';

import Header from 'components/Header';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    app: {},
    dispatch: mockDispatch,
    location: {
      pathname: '/',
    },
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    wrapper.find('.app__header__logo').simulate('click');
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: '@@router/CALL_HISTORY_METHOD',
      payload: { method: 'push', args: [{ pathname: '/', search: undefined, state: undefined }] },
    });

    wrapper.find('.app__logout').simulate('click');
    expect(mockDispatch.mock.calls[1][0]).toEqual({ type: 'USER_LOGOUT_REQUEST' });
  });
});

