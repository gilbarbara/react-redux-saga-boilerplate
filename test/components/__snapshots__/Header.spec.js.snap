// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `"<header class=\\"app__header\\"><div class=\\"app__container\\"><a href=\\"/\\" class=\\"app__header__logo\\"><div class=\\"app__logo\\"><span class=\\"isvg failed\\"><img src=\\"\\" alt=\\"react-redux-saga-boilerplate\\"></span></div></a><div class=\\"app__header__menu\\"><ul class=\\"list-unstyled\\"><li><a href=\\"#logout\\" class=\\"app__logout\\"><span>logout</span><i class=\\"i-sign-out\\"></i></a></li></ul></div></div></header>"`;
